### ISSUE TITLE
Error showing no module named imp

### ISSUE THREAD

**Issue by kaustubhnair02** [PAIR-code/lit#1592]
While trying to run this code on jupyter notebook an error occurred in the line:from lit_nlp import notebook . It further showed that:
File ~/Desktop/LIT_test/venv/lib/python3.12/site-packages/ml_collections/config_flags/config_flags.py:19
     [16](https://file+.vscode-resource.vscode-cdn.net/Users/kaustubhnair/Desktop/LIT_test/~/Desktop/LIT_test/venv/lib/python3.12/site-packages/ml_collections/config_flags/config_flags.py:16) """Configuration commmand line parser."""
     [18](https://file+.vscode-resource.vscode-cdn.net/Users/kaustubhnair/Desktop/LIT_test/~/Desktop/LIT_test/venv/lib/python3.12/site-packages/ml_collections/config_flags/config_flags.py:18) import errno
---> [19](https://file+.vscode-resource.vscode-cdn.net/Users/kaustubhnair/Desktop/LIT_test/~/Desktop/LIT_test/venv/lib/python3.12/site-packages/ml_collections/config_flags/config_flags.py:19) import imp
     [20](https://file+.vscode-resource.vscode-cdn.net/Users/kaustubhnair/Desktop/LIT_test/~/Desktop/LIT_test/venv/lib/python3.12/site-packages/ml_collections/config_flags/config_flags.py:20) import os
     [21](https://file+.vscode-resource.vscode-cdn.net/Users/kaustubhnair/Desktop/LIT_test/~/Desktop/LIT_test/venv/lib/python3.12/site-packages/ml_collections/config_flags/config_flags.py:21) import re

ModuleNotFoundError: No module named 'imp'.
What might be the issue because it is running fine with google colab but having issue with jupyter notebook.
![Screenshot 2024-11-06 at 10 09 32 PM](https://github.com/user-attachments/assets/b2b6aedd-338d-4415-9ef5-15b601e187cc)

**Comment by bdu91** [comment]
It seems a problem with [python 3.12](https://stackoverflow.com/questions/77274572/multiqc-modulenotfounderror-no-module-named-imp), can you try 3.11 and see if it works?

**Comment by RyanMullins** [comment]
Yes, `imp` was [removed](https://docs.python.org/3.11/library/imp.html) in Py3.12 and the `ml-collections` package [knew about and fixed it](https://github.com/google/ml_collections/issues/24) on GitHub back in Jun 2024, but they [haven't published an update](https://pypi.org/project/ml-collections/#history) on PyPI since Jan 2022. 

Short term, you can get around this by reinstalling `ml-collecitons` [from source](https://pip.pypa.io/en/stable/topics/vcs-support/#git) with `pip`:

```sh
pip install -U https://github.com/google/ml_collections.git@2699256
```

Longer term we need `ml-collections` to publish an update on PyPI. Luckily, `ml-collections` is also a Google-owned project, so I can reach out to the maintainers internally and hopefully get this sorted quickly.

**Comment by RyanMullins** [comment]
FYI: I've filed an internal bug to update `ml-collections` on PyPI. I will update and close this issue once the pakcage has been updated.

**Comment by RyanMullins** [comment]
The `ml-collections` team has published v1.0.

https://pypi.org/project/ml-collections/

I'm not aware of any other transient deps on `imp`, so I'm going to close this issue, but please re-open it if you run into any others.
